gflownet/envs/crystals.py

```diff
@@ -139,68 +152,33 @@ def get_mask_invalid_actions(self, state=None, done=None):
                     mask[idx] = True
         return mask
 
-    def state2oracle(self, state_list):
+    def state2oracle(self, state: List = None) -> Tensor:
         """
-        Prepares a list of states in "GFlowNet format" for the oracles: a list of length
-        n_dim with values in the range [cell_min, cell_max] for each state.
+        Prepares a list of states in "GFlowNet format" for the oracle
 
         Args
         ----
-        state_list : list of lists
-            List of states.
-        """
-        return [
-            (
-                self.state2obs(state).reshape((self.n_dim, self.length))
-                * self.cells[None, :]
-            ).sum(axis=1)
-            for state in state_list
-        ]
-
-    def state2obs(self, state=None):
-        """
-        Transforms the state given as argument (or self.state if None) into a
-        one-hot encoding. The output is a list of len length * n_dim,
-        where each n-th successive block of length elements is a one-hot encoding of
-        the position in the n-th dimension.
+        state : list
+            A state
 
-        Example:
-          - State, state: [0, 3, 1] (n_dim = 3)
-          - state2obs(state): [1, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0] (length = 4)
-                              |     0    |      3    |      1    |
+        Returns
+        ----
+        oracle_state : Tensor
+            Tensor containing # of Li atoms, total # of atoms, and fractions of individual elements
         """
         if state is None:
-            state = self.state.copy()
-
-        # z = np.zeros(self.obs_dim, dtype=np.float32)
+            state = self.state
 
-        if len(state) > 0:
-            if hasattr(
-                state[0], "device"
-            ):  # if it has a device at all, it will be cuda (CPU numpy array has no dev
-                state = [subseq.cpu().detach().numpy() for subseq in state]
+        li_idx = self.elem2idx.get(3)
```

Comment:
Just wanted to confirm that state2oracle will be Li or "current task" specific?

Reply:
For now that's the case (I thought that's what we agreed on? But maybe I remembered wrong, let me know), the alternative would be to just return the state and do this conversion in a proxy/oracle wrapper.

Reply:
I'm happy to keep it simple like this for now. We can always update it later when/if we move to non-Li compounds.